openassessment/assessment/api/peer.py

```diff
@@ -103,8 +103,8 @@ def submitter_is_finished(submission_uuid, peer_requirements):
         return False
     except PeerWorkflow.DoesNotExist:
         return False
-    except KeyError:
-        raise PeerAssessmentRequestError(u'Requirements dict must contain "must_grade" key')
+    except KeyError as ex:
```

Comment:
I'm not familiar with this, what does this change do?

Reply:
This is something recent versions of `pylint` have started enforcing, that we use Python 3’s exception chaining, in short, not using `raise from` makes the traceback inaccurate, because the message implies there is a bug in the exception-handling code itself, which is a separate situation than wrapping an exception. When we use `raise from`, the `__cause__` attribute is set, stating the original exception in traceback.